File: src/#237/solution.js
function ListNode(val) {
  this.val = val
  this.next = null
}

var deleteNode = function (node) {
    let currVal=node.val
    let nextVal=node.next.val

    let temp=currVal
    node.val=nextVal
    node.next.val=temp
    node.next=node.next.next

}

File: src/#237/solution.go
package solution

type ListNode struct {
	Val  int
	Next *ListNode
}

func deleteNode(node *ListNode) {
	currVal := node.Val
	nextVal := node.Next.Val

	temp := currVal
	node.Val = nextVal
	node.Next.Val = temp
	node.Next = node.Next.Next
}
